pkg/storage/utils/eosfs/eosfs.go

```diff
@@ -520,36 +520,88 @@ func (fs *eosfs) UnsetArbitraryMetadata(ctx context.Context, ref *provider.Refer
 }
 
 func (fs *eosfs) AddGrant(ctx context.Context, ref *provider.Reference, g *provider.Grant) error {
+	u, err := getUser(ctx)
+	if err != nil {
+		return errors.Wrap(err, "eosfs: no user in ctx")
+	}
+
 	p, err := fs.resolve(ctx, ref)
 	if err != nil {
 		return errors.Wrap(err, "eosfs: error resolving reference")
 	}
 	fn := fs.wrap(ctx, p)
 
+	auth, err := fs.getUserAuth(ctx, u, fn)
+	if err != nil {
+		return err
+	}
+
+	rootAuth, err := fs.getRootAuth(ctx)
+	if err != nil {
+		return err
+	}
+
+	// position where put the ACL
+	position := eosclient.StartPosition
+
+	eosACL, err := fs.getEosACL(ctx, g)
+	if err != nil {
+		return err
+	}
+
+	err = fs.c.AddACL(ctx, auth, rootAuth, fn, position, eosACL)
+	if err != nil {
+		return errors.Wrap(err, "eosfs: error adding acl")
+	}
+	return nil
+
+}
+
+func (fs *eosfs) DenyGrant(ctx context.Context, ref *provider.Reference, g *provider.Grantee) error {
```

Comment:
We also need to check if the caller has the permission to deny access to the resource by checking the `DenyGrant` bit